pkg/test/utils/kubernetes_test.go

```diff
@@ -97,3 +98,69 @@ func TestCreateOrUpdate(t *testing.T) {
 		quit <- true
 	}
 }
+
+func TestGETAPIResource(t *testing.T) {
+	fake := FakeDiscoveryClient()
+
+	for _, test := range []struct {
+		testName string
+		resource runtime.Object
+		found    bool
+	}{
+		{"namespaced resource", NewPod("hello", ""), true},
+		{"namespace already set", NewPod("hello", "other"), true},
+		{"not-namespaced resource", NewResource("v1", "Namespace", "hello", ""), true},
+		{"non-existent resource", NewResource("v1", "Blah", "hello", ""), false},
+	} {
+		t.Run(test.testName, func(t *testing.T) {
+			gvk := test.resource.GetObjectKind().GroupVersionKind()
```

Comment:
Sorry, I don't understand this test :( I have no idea why some resource exist and some do not. Is there some shared state for the whole suite? Can't we do that explicitly? For every test create what that test needs and delete it after it finishes. I find that test design much easier to follow and even maintain...

Reply:
I will add some comments to the test, but the resources are defined in the strict above. It is checking if the types exist in the kubernetes API: a Pod, a Namespace, and a make believe type a “Blah”. The types are added to the FakeDiscoveryClient in the method above. There is no shared state - we aren’t checking if the resources actually exist, just their types.

Reply:
Okay, I gutted the test parameterization. I think it should be a lot more clear now.